distributed/dashboard/components/scheduler.py

```diff
@@ -2036,37 +2073,35 @@ def update(self):
                 nodes_data["node_line_width"].append(1)
 
             # progress bar data update
-            Lbar = self.width_node
-            nodes_data["x_start"].append(x - self.width_node / 2)
-            nodes_data["x_end"].append(x - self.width_node / 2 + Lbar)
+            Lbar = width_box
+            nodes_data["x_start"].append(x - width_box / 2)
+            nodes_data["x_end"].append(x - width_box / 2 + Lbar)
 
-            Hbar = self.height_node * 0.4
-            nodes_data["y_start"].append(y - self.height_node / 2)
-            nodes_data["y_end"].append(y - self.height_node / 2 + Hbar)
+            Hbar = height_box * 0.4
+            nodes_data["y_start"].append(y - height_box / 2)
+            nodes_data["y_end"].append(y - height_box / 2 + Hbar)
 
             completed = comp_tasks / tot_tasks
 
             nodes_data["progress"].append(completed * 100)
-            nodes_data["x_end_progress"].append(
-                x - self.width_node / 2 + Lbar * completed
-            )
+            nodes_data["x_end_progress"].append(x - width_box / 2 + Lbar * completed)
 
             # comp/tot label
             nodes_data["xc_pbar"].append(x)
-            nodes_data["yc_pbar"].append(y - self.height_node / 2 + Hbar / 2)
+            nodes_data["yc_pbar"].append(y - height_box / 2 + Hbar / 2)
 
             nodes_data["comp_tasks"].append(f"{comp_tasks}/{tot_tasks}")
 
             # arrows
             arrows_data["xs"] += [
-                self.nodes_layout[k]["x"] + self.width_node / 2
+                self.nodes_layout[k]["x"] + width_box / 2
```

Comment:
> Boxes now change shapes, still need to fix scaling. But somehow this is causing a problem with the 
 arrows drawing x-position (which depends on the width). It seems it's not drawing the most recent update or not updating properly.

The problem is right here, after talking to  @jrbourbeau the start and end of arrows depend on the dependencies of the current TG. Now `width_box` is the width of the current TG (being updated) but what we need here is the width of the tasks in the dependencies we are looping over. 
To be able to do this we would need to pre-compute the width of every box and assign it to a dict, that we will use when computing the coords of start and end of the arrows.
It will be something like:

```python 
arrows_data["xs"] += [
                self.nodes_layout[k]["x"] + width_box[k] / 2
                for k in self.arrows_layout[key]["nstart"]
            ]
            
arrows_data["xe"] += [
                self.nodes_layout[k]["x"] - width_box[k] / 2
                for k in self.arrows_layout[key]["nend"]
            ]
```